server/routes/index.ts

```diff
@@ -82,6 +84,12 @@ export function defineRoutes(router: IRouter) {
     inputSchema.validate(requestBody); // throws error if validation fail
   }
 
+  function validateEntityId(resourceName: string) {
+    if (!RESOURCE_ID_REGEX.test(resourceName)) {
+      return 'Invalid entity name or id.';
```

Comment:
1. Do we want to handle the empty?
2. Do we want to tell user the valid char set?

Reply:
we don't allow empty the regular express requires some letter to match

And we have hint on the UI to tell users what characters are allowed, so I don't plan to to implement logic to identity invalid characters

Reply:
Agreed.

Reply:
Agreed.

Reply:
updated the regular expression to include unicode characters

Reply:
Curious about the reason not using builtin regex like `/\p{L}+/u`? What is the difference?
And could we apply the length limit 2~50 here? BTW a unicode counts as 1 char or 2 chars?